tool/teleport/common/teleport.go

```diff
@@ -477,6 +483,14 @@ func onConfigDump(flags dumpFlags) error {
 		return trace.Wrap(err)
 	}
 
+	entries, err := os.ReadDir(flags.DataDir)
+	if !os.IsNotExist(err) && len(entries) != 0 {
```

Comment:
Not sure I understand this condition. If the intention is to check that this is existing non-empty directory, should not it be `if err == nil && len(entries) != 0`?

Reply:
It looks like teleport creates the directory if it doesn't exist (see https://github.com/gravitational/teleport/blob/master/lib/service/service.go#L641), that's why I only check if it is empty in case it doesn't
 exist. But this is hard to understand, thoughts on checking just `if len(entries) != 0 `?

Reply:
Fixed in https://github.com/gravitational/teleport/pull/11766/commits/ced98ff7ac9cb9a5e7b6d67384fe22ec7a124197, thanks